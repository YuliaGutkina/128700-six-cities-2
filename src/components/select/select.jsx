import React from 'react';
import PropTypes from "prop-types";


export const Select = (props) => {
  const {onChange, defaultValue, options, className, optionClassName, id} = props;

  const _handleChange = (e) => {
    onChange(e.target.value);
  };

  return <select
    onChange={_handleChange}
    value={defaultValue}
    className={className}
    id={id}
  >
    {options.map((option) => <option
      key={option.value}
      className={optionClassName}
      value={option.value}>
      {option.text}
      t</option>
    )}
  </select>;
};

Select.propTypes = {
  onChange: PropTypes.func.isRequired,
  defaultValue: PropTypes.string,
  className: PropTypes.string,
  optionClassName: PropTypes.string,
  id: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.object),
};
