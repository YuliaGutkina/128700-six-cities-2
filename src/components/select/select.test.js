import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';

import {Select} from "./select";


it(`Select correctly renders after relaunch`, () => {
  const renderer = new ShallowRenderer();
  renderer.render(
      <Select
        onChange={jest.fn}
        value={`popular`}
        options={[{value: `1`, text: `fake1`}, {value: `2`, text: `fake2`}]}
        id="0"
        className="test"
        optionClassName="test-option"
      />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
