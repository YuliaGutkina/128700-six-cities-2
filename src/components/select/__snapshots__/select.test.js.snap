// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select correctly renders after relaunch 1`] = `
<select
  className="test"
  id="0"
  onChange={[Function]}
>
  <option
    className="test-option"
    value="1"
  >
    fake1
  </option>
  <option
    className="test-option"
    value="2"
  >
    fake2
  </option>
</select>
`;
